packages/grid/x-data-grid/src/components/cell/GridEditBooleanCell.tsx

```diff
@@ -92,7 +92,7 @@ function GridEditBooleanCell(props: GridEditBooleanCellProps) {
 
   useEnhancedEffect(() => {
     if (hasFocus) {
-      inputRef.current!.focus();
+      inputRef.current?.focus();
```

Comment:
@cherniavskii I don't think the data grid implementation should change. Instead, we should make the Joy UI slot work.